scipy/optimize/_minimize_constrained.py

```diff
@@ -0,0 +1,485 @@
+
+        where x is an array with shape (n,).
+    x0 : ndarray, shape (n,)
+        Initial guess. ``len(x0)`` is the dimensionality of the minimization
+        problem.
+    method : {str, None}, optional
+        Type of solver. Should be on of:
+
+            - 'equality-constrained-sqp'
+            - 'tr-interior-point'
+
+        When ``None`` the more appropriate method is choosen.
+        'equality-constrained-sqp' is chosen for problems that
+        only have equality constraints and 'tr-interior-point'
+        for general optimization problems.
+    constraints : Constraint or List of Constraint's, optional
+        A single object or a list of objects specifying
+        constraints to the optimization problem.
+        Available constraints are:
+
+            - `BoxConstraint`
+            - `LinearConstraint`
+            - `NonlinearConstraint`
+
+    xtol : float, optional
+        Tolerance for termination by the change of the independent variable.
+        The algorithm will terminate when ``delta < xtol``, where ``delta``
+        is the algorithm trust-radius. Default is 1e-8.
+    gtol : float, optional
+        Tolerance for termination by the norm of the lagrangian gradient.
+        The algorithm will terminate when both the infinite norm of the
+        lagrangian gradient and the constraint violation are smaller than
+        ``gtol``. Default is 1e-8.
+    sparse_jacobian : {boolean, None}
+        The algorithm uses a sparse representation of the Jacobian if True
+        and a dense representation if False. When ``None`` the algorithm
+        uses the more convenient option (which is not always the more effective
+        one).
+    options : dict, optional
+        A dictionary of solver options. Available options include:
+
+            initial_trust_radius: float
+                Initial trust-region radius. Default is 1.
+            initial_penalty : float
+                Initial penalty for merit function. Default is 1.
+            initial_barrier_parameter: float
+                Initial barrier parameter. Exclusive for ``tr_interior_point``
+                method. By defaut uses 0.1.
+            initial_tolerance: float
+                Initial subproblem tolerance. Exclusive for
+                ``tr_interior_point`` method. By defaut uses 0.1.
+            return_all : bool, optional
+                When ``true`` return the list of all vectors
+                through the iterations.
+            factorization_method : string, optional
+                Method used for factorizing the jacobian matrix.
+                Should be one of:
+
+                - 'NormalEquation': The operators
+                   will be computed using the
+                   so-called normal equation approach
+                   explained in [1]_. In order to do
+                   so the Cholesky factorization of
+                   ``(A A.T)`` is computed. Exclusive
```

Comment:
I have seem this notation used in other parts of scipy and I particularly like it. It is easy to get what I am saying (since it is similar to the mathematical notation) and it is more compact than numpy notation.